extensions/db/src/main/java/org/mule/extension/db/internal/DbConnector.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.db.internal;
+
+import org.mule.extension.db.api.StatementStreamingResultSetCloser;
+import org.mule.extension.db.api.param.CustomDataType;
+import org.mule.extension.db.api.param.JdbcType;
+import org.mule.extension.db.api.param.QueryDefinition;
+import org.mule.extension.db.api.param.StoredProcedureCall;
+import org.mule.extension.db.internal.domain.connection.DbConnectionProvider;
+import org.mule.extension.db.internal.domain.connection.derby.DerbyConnectionProvider;
+import org.mule.extension.db.internal.domain.connection.mysql.MySqlConnectionProvider;
+import org.mule.extension.db.internal.domain.type.ArrayResolvedDbType;
+import org.mule.extension.db.internal.domain.type.CompositeDbTypeManager;
+import org.mule.extension.db.internal.domain.type.DbType;
+import org.mule.extension.db.internal.domain.type.DbTypeManager;
+import org.mule.extension.db.internal.domain.type.MappedStructResolvedDbType;
+import org.mule.extension.db.internal.domain.type.MetadataDbTypeManager;
+import org.mule.extension.db.internal.domain.type.ResolvedDbType;
+import org.mule.extension.db.internal.domain.type.StaticDbTypeManager;
+import org.mule.extension.db.internal.operation.DmlOperations;
+import org.mule.runtime.core.api.lifecycle.Initialisable;
+import org.mule.runtime.core.api.lifecycle.InitialisationException;
+import org.mule.runtime.core.util.collection.ImmutableListCollector;
+import org.mule.runtime.extension.api.annotation.Export;
+import org.mule.runtime.extension.api.annotation.Extension;
+import org.mule.runtime.extension.api.annotation.Operations;
+import org.mule.runtime.extension.api.annotation.Parameter;
+import org.mule.runtime.extension.api.annotation.connector.Providers;
+import org.mule.runtime.extension.api.annotation.dsl.xml.Xml;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Connector for connecting to relation Databases through the JDBC API
+ *
+ * @since 4.0
+ */
+@Extension(name = "DB Connector", description = "Connector for connecting to relation Databases through the JDBC API")
+@Operations({DmlOperations.class})
+@Providers({DbConnectionProvider.class, DerbyConnectionProvider.class, MySqlConnectionProvider.class})
+@Xml(namespace = "dbn")
```

Comment:
The module namespace is the `@Extension(name=)`. This is for cases in which you need to force an XML namespace different from the module namespace (default). WHen we delete the old DB connector, then this annotation will go away